src/main/java/mate/academy/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,108 @@
+            statement.setDouble(2, element.getPrice().doubleValue());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                element.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Data processing exception", e);
+        }
+        return element;
+    }
+
+    @Override
+    public Product update(Product element) {
+        String query = "UPDATE products SET productname = ?, productprice = ? "
+                + "WHERE product_id = ? AND deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, element.getName());
+            statement.setDouble(2, element.getPrice().doubleValue());
+            statement.setLong(3, element.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Data processing exception", e);
+        }
+        return element;
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        String query = "SELECT * FROM products WHERE product_id = ? AND deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                return Optional.of(retrieveProductFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Data processing exception", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Product> getAll() {
+        String query = "SELECT * FROM products WHERE deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            List<Product> productList = new ArrayList<>();
+            while (resultSet.next()) {
+                productList.add(retrieveProductFromResultSet(resultSet));
+            }
+            return productList;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Data processing exception", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE products SET deleted = true WHERE product_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            statement.executeUpdate();
+            return true;
```

Comment:
should we always return true? let's check the result of statement.executeUpdate() and return the result of checking